stan/math/opencl/matrix_cl.hpp

```diff
@@ -220,9 +220,11 @@ class matrix_cl<T, require_arithmetic_t<T>> {
   /**
    * Constructor for the matrix_cl that creates a copy of a std::vector of Eigen
    * matrices on the OpenCL device. Each matrix is flattened into one column
-   * of the resulting matrix_cl.
+   * of the resulting matrix_cl. If a lvalue is passed to this constructor the
+   * caller must make sure that the vector does not go out of scope before
```

Comment:
@t4c1, first off, thank you for adding information that there is a requirement on the caller to handle memory.

The documentation isn't quite there yet. It's not clear from reading this when "copying is complete." If I'm trying to understand the requirements on memory management, it would help to be more precise about what this means.

Reply:
Hopefully it is clear now.